shell/platform/darwin/ios/rendering_api_selection.mm

```diff
@@ -24,7 +27,6 @@ bool ShouldUseMetalRenderer() {
   if (@available(iOS METAL_IOS_VERSION_BASELINE, *)) {
     auto device = MTLCreateSystemDefaultDevice();
     ios_version_supports_metal = [device supportsFeatureSet:MTLFeatureSet_iOS_GPUFamily1_v3];
-    [device release];
```

Comment:
This kicked off a yak shave to remove this code by removing `SHELL_ENABLE_METAL`

https://github.com/flutter/engine/pull/51636/files#diff-1db60a8a74c1cfb40c970541dd1f7f0f6301853a2ba17451f5117f154036d732L27